src/Graphics/Image/Processing/Filter.hs

```diff
@@ -207,5 +209,25 @@ gaussianSmoothingFilter !border =
                           ,[  4, 16, 26, 16, 4 ]
                           ,[ 1, 4, 7, 4, 1 ]]
 
-{-# INLINE gaussianSmoothingFilter #-}       
+{-# INLINE gaussianSmoothingFilter #-}    
+
+
+-- | The mean filter is a simple sliding-window spatial filter that replaces the 
+-- center value in the window with the average (mean) of all the pixel values in 
+-- the window. The window, or kernel, is usually square but can be any shape.  
```

Comment:
This statement applies to mean filter in general:
> The window, or kernel, is usually square but can be any shape. 

But it is better when function documentation talks about the actual implementation, so I think it is important to say that this particular filter has a square 3x3 kernel. Maybe something like that:
> The window, or kernel, can be any shape, but this one uses most common 3x3 square kernel. 

Or something along those lines.